C#8 simplified using to reduce nesting

gerhardol opened the pull request:
Follow up to #8729

## Proposed changes

Some using statements missed in #8729 applied (using Intellisense suggestions only, no manual changes)
The second commit makes manual changes, to also use simplified using also if return is not in the using (so scope is marginally expanded).

## Test methodology <!-- How did you ensure quality? -->

Manual

----

:black_nib: I contribute this code under [The Developer Certificate of Origin](../blob/master/contributors.txt).

drewnoakes commented:
I had started annotating the GitUI project. This is going to conflict pretty heavily. I hope to finish up that work this week.

Was this an automated refactoring? It should be, as far as I know. If so, can we hold off on merging this and do it again after I get the GitUI null annotations in?

gerhardol commented:
I can rebase this and #8789 on your PR (and wait with further #8788 PRs) until you are done.

#8789 is regex only (maybe a few manual touchups).
The first commit in this PR is invoking the Intellisense action manually /regex search), not so much time. I will be happy to find how to automate this...
The second commit is manual (return is outside the using{}, so the implementation may differ by a few assembly lines.
All is manageable.

drewnoakes commented:
Thanks @gerhardol. I'll try and finish up that PR this week. Turns out that annotating the GitUI assembly was quite challenging.

RussKie commented on GitUI/GitUICommands.cs:
```diff
@@ -1049,7 +977,7 @@ public bool StartRemotesDialog(IWin32Window owner, string preselectRemote = null
             {
-                using (var form = new FormRemotes(this))
+                using var form = new FormRemotes(this);
                 {
```
Either `{ }` or `;` is incorrect here.

gerhardol replied:
This was a manual change...
Will squash after drew is done

RussKie reviewed (commented):
👍

drewnoakes commented:
@gerhardol just to say that I am still working on this. I'm down to 59 annotation errors in 8 files within the GitUI project. The last ones are tricky, but I'm getting there. There's been some very loose and wild use of null throughout the code. It's satisfying to map this all out as a first step towards tightening some of these usages up.

gerhardol commented:
> @gerhardol just to say that I am still working on this.

Thanks, I assumed that, annotation is a big change. I will see when you are ready!

gerhardol commented:
squashed comments and rebased after the GitUI annotations PR was merged. Some manual changes due to conflicts. Will squash the IntelliSense and manual commit at next update.
Drew has another annotations PR, I guess that should go first.

drewnoakes reviewed (approved):
Don't hold this up for #8878. It doesn't look like there'll be much in the way of conflicts. I will deal with any that arise 👍

gerhardol commented:
squashed and rebased, no further changes

gerhardol commented:
@msftbot merge in 24 hours
